Add units to sensor log API request model

samcorky opened the pull request:
Not going into DB - this will be used in SMIB > v2.0 
 Adding now to allow SMIBHID development

samcorky committed:
Add units to sensor log API request model
Not going into DB - this will be used in SMIB > v2.0
Adding now to allow SMIBHID development

sjefferson99 commented:
@sam57719 See note in smib issue, can we restructure more like that? Will wait to confirm before building the SMIBHID implementation.

samcorky commented:
Happy with the proposed changes! Will implement tomorrow.

sjefferson99 commented:
I'll build something now and test tomorrow when SMIB is updated.

samcorky committed:
Add units field to sensor log payload structure

Refactored the sensor log API model to include a units field alongside readings. Updated the payload structure and test stub to reflect these changes. This prepares for future SMIB > v2.0 updates while enabling consistent SMIBHID development.

samcorky commented:
SMIB Changed in this PR to accept data like:
```json
{
    "readings": [
        {
            "data": {
                "BME280": {
                    "humidity": 48.7,
                    "pressure": 1019.3,
                    "temperature": 20.03
                },
                "SCD30": {
                    "co2": 0,
                    "relative_humidity": 0,
                    "temperature": 0
                }
            },
            "human_timestamp": "2025-07-16T17:27:54.322400+00:00",
            "timestamp": 1752686874
        },
        {
            "data": {
                "BME280": {
                    "humidity": 57.64,
                    "pressure": 632,
                    "temperature": 23.05
                },
                "SCD30": {
                    "co2": 1548.1,
                    "relative_humidity": 52.9,
                    "temperature": 26.3
                }
            },
            "human_timestamp": "2025-07-16T17:27:54.322400+00:00",
            "timestamp": 1752686874
        }
    ],
    "units": {
        "BME280": {
            "humidity": "%",
            "pressure": "hPa",
            "temperature": "C"
        },
        "SCD30": {
            "co2": "ppm",
            "relative_humidity": "%",
            "temperature": "C"
        }
    }
}
```

samcorky committed:
Simplify payload creation in stub_sensor_log

Replaced verbose hardcoded structures with concise inline JSON. This improves clarity and reduces unnecessary code duplication for test fixtures.

sjefferson99 reviewed (approved):
Confirmed payload from smibhid successfully logs.
